src/pages/map/index.tsx

```diff
@@ -59,19 +78,21 @@ const LandmarkTabPanel = ({ latitude }: { latitude: Latitude }) => {
         <SimpleGrid spacing={componentSpacing} 
         templateColumns='repeat(auto-fill, minmax(200px, 1fr))'>
           {data?.map((landmark, index) => (
-            <LandmarkCard key={index} landmark={landmark} />
+            <LandmarkCard key={index} landmark={landmark} openDrawer={openDrawer}/>
           ))}
         </SimpleGrid>
       }
     </>
   );
 };
 
-const LandmarkCard = ({ landmark }: { landmark: Landmark })  => {
+const LandmarkCard = ({ landmark, openDrawer }: { 
+  landmark: Landmark; openDrawer: (landmark: Landmark) => void  })  => {
   const maxChar: number = useBreakpointValue({ base: mobileTextLimit, 
     [sidebarBreakpoint]: desktopTextLimit }) || desktopTextLimit;
   const cardText = landmark.定义.length <= maxChar ? landmark.定义 : <Text>
-    {landmark.定义.substring(0, maxChar)}...{' '}<Link>更多</Link>
+    {landmark.定义.substring(0, maxChar)}...{' '}
+    <Link onClick={ () => openDrawer(landmark) }>更多</Link>
```

Comment:
Format wise, `{() => openDrawer(landmark)}` -- no space. But you should revert this change. No need to do it here. See next comment.